### PR TITLE
Update controller-runtime and deps

### LINKED ISSUE #549: operator does not patch pod
Hello! After creating a cluster, the operator cannot patch under to hang new labels

kubernetes: 1.17.2

```
{"severity":"ERROR","timestamp":"2020-06-22T11:58:23.03836963Z","logger":"kubebuilder.controller","message":"Reconciler error","controller":"controller.mysqlcluster","request":"mautic/mautic","error":"Pod \"mautic-mysql-0\" is invalid: spec: Forbidden: pod updates may not change fields other than `spec.containers[*].image`, `spec.initContainers[*].image`, `spec.activeDeadlineSeconds` or `spec.tolerations` (only additions to existing tolerations)\n  core.PodSpec{\n  \t... // 20 identical fields\n  \tPriorityClassName: \"high-priority\",\n  \tPriority:          &100,\n- \tPreemptionPolicy:  nil,\n+ \tPreemptionPolicy:  &\"PreemptLowerPriority\",\n  \tDNSConfig:         nil,\n  \tReadinessGates:    nil,\n  \t... // 4 identical fields\n  }\n","stacktrace":"github.com/presslabs/mysql-operator/vendor/github.com/go-logr/zapr.(*zapLogger).Error\n\t/go/src/github.com/presslabs/mysql-operator/vendor/github.com/go-logr/zapr/zapr.go:128\ngithub.com/presslabs/mysql-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/src/github.com/presslabs/mysql-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:217\ngithub.com/presslabs/mysql-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1\n\t/go/src/github.com/presslabs/mysql-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158\ngithub.com/presslabs/mysql-operator/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\t/go/src/github.com/presslabs/mysql-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133\ngithub.com/presslabs/mysql-operator/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/go/src/github.com/presslabs/mysql-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134\ngithub.com/presslabs/mysql-operator/vendor/k8s.io/apimachinery/pkg/util/wait.Until\n\t/go/src/github.com/presslabs/mysql-operator/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88"}
```

### PR DESCRIPTION
**Pull request by AMecea**
closes/fixes #549 

---
 - [x] I've made sure the [Changelog.md](https://github.com/presslabs/mysql-operator/blob/master/Changelog.md) will remain up-to-date after this PR is merged.

### CODE REVIEW

File: pkg/internal/mysqlbackup/mysqlbackup_test.go [hunk 1]
```diff
@@ -16,6 +16,7 @@ limitations under the License.
 package mysqlbackup
 
 import (
+	"github.com/go-logr/zapr"
```
Comment (rstefan1) [inline pkg/internal/mysqlbackup/mysqlbackup_test.go:19]: Please reorder imports

File: pkg/controller/internal/testutil/testutil.go [hunk 2]
```diff
@@ -18,6 +18,8 @@ limitations under the License.
 package testutil
 
 import (
+	"context"
+	"github.com/go-logr/zapr"
```
Comment (rstefan1) [inline pkg/controller/internal/testutil/testutil.go:22]: Reorder imports

File: go.mod [hunk 3]
```diff
@@ -3,30 +3,46 @@ module github.com/presslabs/mysql-operator
...
+	github.com/go-test/deep v1.0.7
+	github.com/gobuffalo/flect v0.2.2 // indirect
 	github.com/golang/glog v0.0.0-20160126235308-23def4e6c14b
-	github.com/imdario/mergo v0.3.8
-	github.com/onsi/ginkgo v1.14.1
-	github.com/onsi/gomega v1.10.1
-
-	github.com/presslabs/controller-util v0.2.3
+	github.com/gophercloud/gophercloud v0.1.0 // indirect
+	github.com/imdario/mergo v0.3.11
```
Comment (rstefan1) [inline go.mod:20]: We should remove `mergo` from deps

**Review by rstefan1 — CHANGES_REQUESTED** [review]

**Review by rstefan1 — APPROVED** [review]

File: go.mod (hunk 3 above)
Comment (rstefan1) [inline go.mod:20]: Don't update `mergo`

**Review by rstefan1 — CHANGES_REQUESTED** [review]
Please update `golangci-lint`

File: go.mod (hunk 3 above)
Comment (AMecea) [reply go.mod]: I've updated it to `0.3.12` where the major bugs are fixed.

**Review by rstefan1 — APPROVED** [review]

**Pull request merged**